File: examples/langchain.py
"""Example: Using claude-code-relay with LangChain."""

from langchain_openai import ChatOpenAI
from langchain_core.messages import HumanMessage, SystemMessage

# Create model pointing to local claude-code-relay server
model = ChatOpenAI(
    model="sonnet",
    openai_api_key="not-needed",
    openai_api_base="http://localhost:52014/v1",
)

# Simple invocation
print("=== Simple ===")
messages = [
    SystemMessage(content="You are a helpful assistant."),
    HumanMessage(content="What is the meaning of life?"),
]
response = model.invoke(messages)
print(response.content)

# Streaming
print("\n=== Streaming ===")
for chunk in model.stream([HumanMessage(content="Tell me a joke.")]):
    print(chunk.content, end="", flush=True)
print()

File: examples/other/langchain.ts
/**
 * Example: Using claude-code-relay with LangChain.js
 */

import { ChatOpenAI } from "@langchain/openai";
import { HumanMessage, SystemMessage } from "@langchain/core/messages";

// Create model pointing to local claude-code-relay server
const model = new ChatOpenAI({
  modelName: "sonnet",
  openAIApiKey: "not-needed",
  configuration: {
    baseURL: "http://localhost:52014/v1",
  },
});

async function main() {
  // Simple invocation
  console.log("=== Simple ===");
  const messages = [
    new SystemMessage("You are a helpful assistant."),
    new HumanMessage("What is the meaning of life?"),
  ];
  const response = await model.invoke(messages);
  console.log(response.content);

  // Streaming
  console.log("\n=== Streaming ===");
  const stream = await model.stream([new HumanMessage("Tell me a joke.")]);
  for await (const chunk of stream) {
    process.stdout.write(String(chunk.content));
  }
  console.log();
}

main().catch(console.error);
